Aztec/Classes/Formatters/AttributeFormatter.swift

```diff
@@ -144,21 +124,34 @@ protocol ParagraphAttributeFormatter: AttributeFormatter {
 
 extension ParagraphAttributeFormatter {
 
-    func applicationRange(for range: NSRange, in string: NSAttributedString) -> NSRange {
-        return string.paragraphRange(for: range)
-    }
-
+    /// Toggles an attribute in the specified range of a text storage, and returns the new Selected Range.
+    ///
+    /// - Note: Whenever either the application paragraph is empty, or the entire storage is empty,
+    ///   we'll need to insert a placeholder (Zero Width String). Reason why? Because some formatters
+    ///   (TextList / Blockquote) need to display a custom UI, even when there is no content to display.
+    ///   In those scenarios, TextView's TypingAttributes just don't do the trick.
+    ///
+    /// - Note: For the reasons mentioned above, the first thing we'll do is to determine if the attribute should
+    ///   be applied or not. Order of events is important. 
+    ///   Why? because we *may need* to insert an empty string placeholder, and this operation may alter this result!
+    ///
     @discardableResult
     func toggle(in text: NSMutableAttributedString, at range: NSRange) -> NSRange? {
-        let applicationRange = self.applicationRange(for: range, in: text)
+        let shouldApply = shouldApplyAttributes(to: text, at: range)
+        var applicationRange = text.paragraphRange(for: range)
         var newSelectedRange: NSRange?
 
         if applicationRange.length == 0 || text.length == 0 {
-            insertEmptyPlaceholderString(in: text, at: applicationRange.location)
+            text.insert(placeholderForAttributedEmptyLine, at: applicationRange.location)
```

Comment:
When inserting this on a empty string, the placeholder attribute will not have the correct paragraph style. We probably should apply to it the default Paragraph Style for Aztec.